Source/Scene/Composite3DTileContentProvider.js

```diff
@@ -0,0 +1,166 @@
+/*global define*/
+define([
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/DeveloperError',
+        '../Core/getStringFromTypedArray',
+        '../Core/loadArrayBuffer',
+        './Cesium3DTileContentState',
+        '../ThirdParty/when'
+    ], function(
+        defaultValue,
+        defined,
+        DeveloperError,
+        getStringFromTypedArray,
+        loadArrayBuffer,
+        Cesium3DTileContentState,
+        when) {
+    "use strict";
+
+    /**
+     * DOC_TBA
+     */
+    var Composite3DTileContentProvider = function(tileset, tile, url, factory) {
+        this._url = url;
+        this._tileset = tileset;
+        this._tile = tile;
+        this._contentProviders = [];
```

Comment:
I'm still not sure about this. I could just add a getter, but then it's tile.content.contents. Not that great. Maybe tile.content could sometime return an array? It's still odd.

The color changing works automatically though without any modifications.